cocos/2d/CCParticleSystem.h

```diff
@@ -251,6 +251,12 @@ class CC_DLL ParticleSystem : public Node, public TextureProtocol, public Playab
      */
     static ParticleSystem* createWithTotalParticles(int numberOfParticles);
     
+private:
+    // Get the reference of all ParticleSystem, it's used by EngineDataManager internally.
+    // Make it public while it's needed, but not now.
+    static std::vector<ParticleSystem*>& getAllParticleSystems();
+    friend class EngineDataManager;
+public:
```

Comment:
Do we just need particle numbers? How about `getAllParticleNumbers()`? And this function can be public too.